### DISCUSSION TITLE
Create Topography with numpy

### DISCUSSION THREAD

**Discussion by anneschulz1991** [gempy-project/gempy#735] [Q&A]
Hello to all,
it's me again and I need help again. I saw on the Gempy website that it is possible to determine the topography using a numpy array. Is there anyone who can send me the exact command and show me how the array should look exactly (with coordinates or without)?
-->geo_model.set_topography(source='numpy'....?
Maybe there is already an example.
Many greetings
Anne

**Comment by anneschulz1991** [comment]
I ask because my tif file keeps giving the following error message.  

ERROR 1: Attempt to create 0x1237 dataset is illegal,sizes must be larger than zero.
ERROR 4: _cropped_DEM.tif: No such file or directory

---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
/tmp/ipykernel_106191/3882126103.py in <module>
----> 1 geo_model.set_topography(source="gdal", filepath="/home/anne/Dokumente/gempy/ExtraktionDaten/NO-SW_Profil_1/Topographie/TOPO_TEST.tif")

~/gempy/gempy/plot/decorators.py in pst(*args, **kwargs)
    106     def pst(*args, **kwargs):
    107         plot_object = kwargs.pop('plot_object') if 'plot_object' in kwargs else None
--> 108         topography = func(*args, **kwargs)
    109 
    110         if plot_object is not None:

~/gempy/gempy/core/model.py in set_topography(self, source, set_mask, **kwargs)
    426         """
    427 
--> 428         self._grid.create_topography(source, **kwargs)
    429         if set_mask is True:
    430             try:

~/gempy/gempy/core/data.py in create_topography(self, source, **kwargs)
    181             filepath = kwargs.get('filepath', None)
    182             if filepath is not None:
--> 183                 self.topography.load_from_gdal(filepath)
    184             else:
    185                 print('to load a raster file, a path to the file must be provided')

~/gempy/gempy/core/grid_modules/topography.py in load_from_gdal(self, filepath)
    120 
    121     def load_from_gdal(self, filepath):
--> 122         dem = LoadDEMGDAL(filepath, extent=self.extent)
    123         self._x, self._y = None, None
    124         self.set_values(dem.get_values())

~/gempy/gempy/core/grid_modules/create_topography.py in __init__(self, path_dem, grid, extent, delete_temp)
     50         if extent is not None:
     51             self.regular_grid_extent = extent
---> 52             self.crop2grid()
     53         elif grid is not None:
     54             self.regular_grid_extent = grid.extent

~/gempy/gempy/core/grid_modules/create_topography.py in crop2grid(self, delete_temp)
    103 
    104             self.dem = gdal.Open(path_dest)
--> 105             self.dem_zval = self.dem.ReadAsArray()
    106             self._get_raster_dimensions()
    107 

AttributeError: 'NoneType' object has no attribute 'ReadAsArray'

The file path is definitely correct. And the fix for ERROR 4 was unsuccessful. So I wanted to try it via an array.

**Reply by 2651666L** [reply]
Hello,
I have this problem, too. Have you solve the problem? I don't know is it because of the property of the tif file or something else. The file in tutorial can be read correctly but for my own file it will come to this problem.

**Reply by AlexanderJuestel** [reply]
@2651666L Can you open the raster using the rasterio library?

**Comment by javoha** [comment]
I think that fits the expertise of @AlexanderJuestel.

**Comment by AlexanderJuestel** [comment]
@anneschulz1991 and @2651666L, do you still have issues with your rasters using the latest version of GemPy? (2.3)

**Comment by AlexanderJuestel** [comment]
Maybe there is a solution here: https://github.com/cgre-aachen/gempy/discussions/790
